tools/reproduce_comparison/Jenkinsfile

```diff
@@ -1,56 +1,139 @@
-        string(name: 'excludeFiles', defaultValue: 'classes_nocoops.jsa', description: 'File names to be excluded while doing the comparison, separated by `;`')
+        string(name: 'COMPARED_JOB_NUMBER', defaultValue: '', description: 'Compared nightly build job name')
+        string(name: 'COMPARED_JOB_NAME', defaultValue: '', description: 'Compared nightly build job number')
+        string(name: 'COMPARED_JOB_PARAMS', defaultValue: '', description: 'Compared nightly build job parameters')
     }
 
     stages {
-        stage('Lets Compare') {
-            agent {
-                label "${params.platform}&&build&&x64" // build label enables comparisons to be done, set to x64 for linux, might need adjust for windows/mac
-            }
+        stage('Prepare') { //Copy artifacts, reset parameters,trigger build and copyArtifacts
             steps {
+                cleanWs()
+                checkout scm
+                copyArtifacts excludes: '**/OpenJDK*-sbom*metadata.json', 
+                                filter: '**/OpenJDK*-jdk*.tar.gz,**/OpenJDK*-sbom*.json',
+                                fingerprintArtifacts: true, 
+                                flatten: true, 
+                                projectName: "${params.COMPARED_JOB_NAME}",
+                                target: 'original/'
+                                selector: specific("${params.COMPARED_JOB_NUMBER}")
                 script {
-                    try {
-                        cleanWs()
-                        // use Jenkins crendential to download JDK if source is from openjdkX-pipline
-                        withCredentials([usernamePassword(credentialsId: 'eclipse_temurin_bot_email_and_token', passwordVariable: 'PASSWORD', usernameVariable: 'USERNAME')]) {
-                            echo "Fetching artifact1 from ${params.URL1}"
-                            def ret1 = sh returnStatus: true, script: "mkdir url1Dir1 && curl -s --user ${USERNAME}:${PASSWORD} ${params.URL1} | tar -xz  -C url1Dir1"
-                            if (ret1 != 0) {
-                                currentBuild.result = 'UNSTABLE'
-                                error 'Stopping after download and uncompress tar file'
-                            }
-                            echo "Fetching artifact2 from ${params.URL2}"
-                            def ret2 = sh returnStatus: true, script: "mkdir url1Dir2 && curl -s --user ${USERNAME}:${PASSWORD} ${params.URL2} | tar -xz  -C url1Dir2"
-                            if (ret2 != 0) {
-                                currentBuild.result = 'UNSTABLE'
-                                error 'Stopping after download and uncompress tar file'
+                    def sbomFiles = findFiles(glob: "**/*.json")
+                    def sbomParams = readJSON file: "${sbomFiles[0].name}"
+                    def buildTimeStamp = sbomParams.metadata.timestamp
```

Comment:
Rather confusing I know, but sbomParams.metadata.timestamp is the CycloneDX sbom update "timestamp", not the Build timestamp, you need to get "Build Timestamp" in your loop below .. from the sbomParams.components[0].properties

Reply:
eg:
```
        {
          "name" : "Build Timestamp",
          "value" : "2023-06-07 03:31:17"
        },
```

Reply:
> Rather confusing I know, but sbomParams.metadata.timestamp is the CycloneDX sbom update "timestamp", not the Build timestamp, you need to get "Build Timestamp" in your loop below .. from the sbomParams.components[0].properties

I was going to double check it with you about that. Thanks!

Reply:
> Note, I suspect you'll need to craft that into ISO8601, it is UTC. Wondering if we should fix the SBOM so it is stored in ISO8601? https://github.com/adoptium/temurin-build/blob/7045470523b810ec64e580139a431b62f505e65d/sbin/build.sh#L791

Sorry, I'm not sure about this note. Current it is in ISO8601 format UTC time [buildTimestamp=$(date -u +"%Y-%m-%d %H:%M:%S") ](https://github.com/adoptium/temurin-build/blob/master/sbin/build.sh#L283). What is the correct time and correct format?

Reply:
yes agree